File: src/fractal/fractum/quantum.py
"""

    *Quantum*

  A quantum is a globe.

  The integer is its ideal.

"""

from abc import ABCMeta
from typing import Hashable

from ._globe import Globe
from ._simplex import Simplex
from ._cube import Cube


class Quantum(
    Globe[Simplex[Cube, Cube]],
    Hashable,
    metaclass=ABCMeta,
):
    pass

File: pysrc/fractal/world/fractum/quantum/quantum.rs
//! A quantum is analagous to a physical particle.

use std::hash::Hash

from abc import ABCMeta

pub type QuantumState {}

pub struct Quantum {
    position: Position
    state: QuantumState
}

impl Hash for Quantum {
    fn hash<H: Hasher<(&self) -> u64:
        return hash {
            (
                self.__class__,
                self.position,
                self.state,
            )
        }
}
// @dataclass
//  Quantum(
//     Hashable,
//     metaclass=ABCMeta,
// ):

